File: module-android-api/src/main/java/com/tainzhi/sample/api/handler/Runnable4.java
package com.tainzhi.sample.api.handler;


import android.os.Handler;
import android.os.Message;

/**
 * @author: tainzhi
 * @mail: qfq61@qq.com
 * @date: 2019-10-26 08:43
 * @description:
 **/

public class Runnable4 implements Runnable {
	Handler handler;
	
	public Runnable4(Handler handler) {
		this.handler = handler;
	}
	
	@Override
	public void run() {
		Message msg = new Message();
		msg.what = Constants.SUB_2_SUB;
		handler.sendMessage(msg);
	}
}

File: module-android-api/src/main/java/com/tainzhi/sample/api/handler/Runnable4.kt
package com.tainzhi.sample.api.handler

import android.os.Handler
import android.os.Message
import android.widget.TextView

/**
 * @author: tainzhi
 * @mail: qfq61@qq.com
 * @date: 2019-10-26 08:43
 * @description:
 */
class Runnable4(var handler: Handler?, val textView: TextView) : Runnable {
    override fun run() {
        textView.append("${Thread.currentThread().name}: send message\n")
        val msg = Message()
        msg.what = Constants.SUB_2_SUB
        handler!!.sendMessage(msg)
    }
}